### PR TITLE
Added scripted controller act and condition sample implementations

### PR DESCRIPTION
**Pull request by HalalUr**
+ Bugfixes and GameClock

### CODE REVIEW

File: src/core/scripted_controller/bite_act.cpp [hunk 1]
```diff
@@ -0,0 +1,77 @@
+#include "bite_act.h"
+#include "core/i_move_component.h"
+#include "core/i_position_component.h"
+#include "../../engine/collision_system.h"
```
Comment (MrPepperoni) [inline src/core/scripted_controller/bite_act.cpp:4]: no need for ../../, :/src is in the include path

File: src/core/scripted_controller/selection_type.h [hunk 2]
```diff
@@ -0,0 +1,34 @@
+#ifndef INCLUDED_SCRIPTEDCONTROLLER_SELECTION_TYPE_H
+#define INCLUDED_SCRIPTEDCONTROLLER_SELECTION_TYPE_H
+
+#include "platform/singleton.h"
+#include "boost/bimap.hpp"
+
+namespace scriptedcontroller {
+
+class SelectionType : public platform::Singleton<SelectionType>
```
Comment (MrPepperoni) [inline src/core/scripted_controller/selection_type.h:9]: enum or autoid, no use in using both at the same time

File: src/core/scripted_controller/turn_to_target_act.cpp [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        direction = atan2( distV.y, distV.x );
+        L2( "clear vision direction!\n" );
+    }
+    else
+    {
+        direction = mPathSystem->GetDirection( actor, *target );
+    }
+    double rads = direction - positionC->GetOrientation();
+    static const double pi = boost::math::constants::pi<double>();
+    while (rads < -pi)
```
Comment (MrPepperoni) [inline src/core/scripted_controller/turn_to_target_act.cpp:62]: fmod

File: src/main/main.cpp [hunk 4]
```diff
@@ -506,6 +507,9 @@ int main( int argc, char* argv[] )
     L1( "client_datas_message type: %d\n", network::ClientDatasMessage::GetType_static() );
     L1( "soldier_properties_message type: %d\n", network::SoldierPropertiesMessage::GetType_static() );
 
+    platform::Clock::Get();
```
Comment (MrPepperoni) [inline src/main/main.cpp:510]: no need for sep. get if you'll call a member in the next line anyway

**Review by MrPepperoni — APPROVED** [review]
Cool. I think the autoid<->enum thing should be removed, using simply enums is cleaner. Or convince me of the benefits of using both.

File: src/main/main.cpp (hunk 4 above)
Comment (HalalUr) [reply src/main/main.cpp:510]: Done

File: src/core/scripted_controller/turn_to_target_act.cpp (hunk 3 above)
Comment (HalalUr) [reply src/core/scripted_controller/turn_to_target_act.cpp:62]: Thx

File: src/core/scripted_controller/selection_type.h (hunk 2 above)
Comment (HalalUr) [reply src/core/scripted_controller/selection_type.h:9]: Currently it's okay. (talked about the change to simple string variant)

File: src/core/scripted_controller/bite_act.cpp (hunk 1 above)
Comment (HalalUr) [reply src/core/scripted_controller/bite_act.cpp:4]: ok

**Pull request merged**